infrastructure/terraform/components/acct/data_iam_policy_document_sso_read_only_table_access.tf

```diff
@@ -0,0 +1,127 @@
+data "aws_iam_policy_document" "sso_read_only_table_access" {
+
+  statement {
+    sid    = "AllowGlueAccess"
+    effect = "Allow"
+
+    actions = [
+      "glue:GetTable",
+      "glue:GetTables",
+      "glue:BatchGetTable",
+      "glue:GetDatabase",
+      "glue:GetDatabases",
+      "glue:GetPartition",
+      "glue:GetPartitions"
+    ]
+
+    resources = concat(
+      [
+        "arn:aws:glue:${var.region}:${var.aws_account_id}:catalog",
+        "arn:aws:glue:${var.region}:${var.aws_account_id}:database/nhs-notify-*-reporting-database",
+        "arn:aws:glue:${var.region}:${var.aws_account_id}:table/nhs-notify-*-reporting-database/request_item_plan_completed_summary",
+        "arn:aws:glue:${var.region}:${var.aws_account_id}:table/nhs-notify-*-reporting-database/request_item_status"
+      ],
+      [
+        for account_id in var.core_account_ids :
+        "arn:aws:glue:${var.region}:${account_id}:catalog"
+      ]
+    )
+  }
+
+  statement {
+    sid    = "AllowAthenaAccess1"
+    effect = "Allow"
+
+    actions = [
+        "athena:GetQueryResults",
+        "athena:GetQueryResultsStream",
+        "athena:GetQueryExecution",
+        "athena:StartQueryExecution",
+        "athena:GetWorkGroup",
+        "athena:GetNamedQuery"
+    ]
+
+    resources = [
+    "arn:aws:athena:${var.region}:${var.aws_account_id}:workgroup/nhs-notify-*-reporting-user"
+    ]
+  }
+  statement {
+    sid    = "AllowAthenaAccess2"
+    effect = "Allow"
+
+    actions = [
+        "athena:GetDatabase",
+        "athena:GetTableMetadata",
+        "athena:GetDataCatalog",
+        "athena:GetTable"
+    ]
+
+    resources = [
+      "arn:aws:athena:${var.region}:${var.aws_account_id}:datacatalog/AWSDataCatalog"
+    ]
+  }
+
+  statement {
+    sid    = "AllowAthenaAccess3"
+    effect = "Allow"
+
+    actions = [
+        "athena:ListDataCatalogs",
+        "athena:ListDatabases",
+        "athena:ListTableMetadata",
+        "athena:ListWorkGroups"
+    ]
+
+    resources = [ "*" ] # Access to List all above is required. Condition keys not supported for these resources.
+  }
+
+  statement {
+    sid    = "AllowS3KMSAccess"
+    effect = "Allow"
+
+    actions = [
+      "kms:Decrypt",
+      "kms:Encrypt",
+      "kms:GenerateDataKey",
+      "kms:GenerateDataKeyWithoutPlaintext",
+      "kms:DescribeKey"
+    ]
+
+    resources = [
+      aws_kms_key.s3.arn # Not sure how to solve this as we can't pattern match because can't use alias...
```

Comment:
@sidnhs @aidenvaines-bjss any suggestions here? I'd like to avoid an output from the reporting component as I'd imagine the pattern should always be acct first, other component second.

Reply:
AFAIK, I think we can use aliases, has issues when its cross account access, but incase we still hve issues we can do what I have tried before to give access via Aliases regardless, to avoid reading from statefile and stuffs, by having htis condition maybe?:
![image](https://github.com/user-attachments/assets/71437b81-5973-462f-920e-2a746a1eaa53)